phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterTableIT.java

```diff
@@ -852,6 +854,37 @@ public void testDropMultipleColumns() throws Exception {
         conn1.close();
     }
 
+    @Test
+    public void testAlterTableOnGlobalIndex() throws Exception {
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.setAutoCommit(false);
+            Statement stmt = conn.createStatement();
+            String tableName = generateUniqueName();
+            String globalIndexTableName = generateUniqueName();
+
+            stmt.execute("CREATE TABLE " + tableName + "(k VARCHAR PRIMARY KEY, v1 VARCHAR, v2 VARCHAR)");
+            stmt.execute("DROP TABLE " + tableName);
+            Admin admin = conn.unwrap(PhoenixConnection.class).getQueryServices().getAdmin();
+            admin.disableTable(TableName.valueOf(tableName));
+            admin.deleteTable(TableName.valueOf(tableName));
+            conn.createStatement().execute("CREATE TABLE " + tableName +
+                " (ID INTEGER PRIMARY KEY, COL1 VARCHAR(10), COL2 BOOLEAN)");
+
+            conn.createStatement().execute("CREATE INDEX " + globalIndexTableName + " on " + tableName + " (COL2)");
+            TableDescriptor originalDesc = admin.getDescriptor(TableName.valueOf(globalIndexTableName));
+            int expectedErrorCode = 0;
+            try {
+                conn.createStatement().execute("ALTER TABLE " + globalIndexTableName + " ADD CF1.AGE INTEGER ");
+                conn.commit();
```

Comment:
Instead of initializing `expectedErrorCode` to zero to ensure later that the exception happened, you can instead call `Assert.fail()` after `conn.commit()` so it fails in case the SQLException does not happen. Then you can assert the exception code inside the catch statement.